File: python/deepstrike/signals/scheduled.py
from __future__ import annotations
from dataclasses import dataclass, field
from typing import Any


@dataclass
class ScheduledPrompt:
    goal: str
    run_at_ms: int                        # epoch ms
    criteria: list[str] = field(default_factory=list)
    metadata: dict[str, Any] = field(default_factory=dict)

    def to_signal(self) -> "RuntimeSignal":
        from deepstrike.signals.types import RuntimeSignal
        return RuntimeSignal(
            source="cron",
            signal_type="job",
            urgency=self.metadata.get("urgency", "normal"),
            payload={"goal": self.goal, "criteria": self.criteria, **self.metadata},
            dedupe_key=f"cron:{self.goal}:{self.run_at_ms}",
        )

File: node/src/signals/scheduled.ts
import type { RuntimeSignal } from "./types.js"

export class ScheduledPrompt {
  constructor(
    public readonly goal: string,
    public readonly runAtMs: number,
    public readonly criteria: string[] = [],
    public readonly metadata: Record<string, unknown> = {},
  ) {}

  toSignal(): RuntimeSignal {
    return {
      source: "cron",
      signalType: "job",
      urgency: "normal",
      dedupeKey: `cron:${this.goal}:${this.runAtMs}`,
      payload: { goal: this.goal, criteria: this.criteria, runAtMs: this.runAtMs, ...this.metadata },
    }
  }
}
